R/glimmaXY.R

```diff
@@ -150,13 +150,14 @@ buildXYData <- function(
 
   # transform status column to (downReg, nonDE, upReg)
   status <- sapply(status, function (x) {
-    switch(as.character(x), "-1"="downReg", "0"="nonDE", "1"="upReg")
+    switch(ifelse(is.na(x), "NA", as.character(x)), 
+           "-1"="downReg", "0"="nonDE", "1"="upReg", "NA"="NA")
```

Comment:
just for readability's sake we can introduce a separate if clause - wdyt?
```suggestion
    if (is.na(x)) {
      return("NA")
    }
    switch(as.character(x), "-1"="downReg", "0"="nonDE", "1"="upReg")
```